### PR TITLE
Fix: Fixed editing name in Profile.js

### LINKED ISSUE #44: Bug: update the username properly
### Describe the bug
Currently the edit profile functionality is not working as expected. The first character does not get deleted on backspace. This has to be modified accordingly in the frotend part so that the user an easily update the username.

### To Reproduce
1. login to osp
2. Click on dashboard
3. select the pencil icon to edit
4. try to edit the username

### Expected behavior
The username should be edited without retention of any previous values

### Screenshots
user updates

### PR DESCRIPTION
**Pull request by saideepesh000**
### Description
Currently, the edit profile functionality is not working as expected. The first character does not get deleted on backspace. This has to be modified accordingly in the frontend part so that the user an easily update the username.

Fixes #44 

### Type of Change:

- Code

**Code/Quality Assurance Only**
- Bugfix (non-breaking change which fixes an issue)

### How Has This Been Tested?
I've tested it locally, the updated name is also reflected in Django backend.

### Checklist:

**Code/Quality Assurance Only**
- [x] My changes generate no new warnings
- [ ] My PR currently breaks something (fix or feature that would cause existing functionality to not work as expected)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been published in downstream modules

### CODE REVIEW

File: src/components/Profile.js [hunk 1]
```diff
@@ -14,6 +14,7 @@ class Profile extends Component {
             error: null,
             submitted: false
         }
+
```
Comment (bismitaguha) [inline src/components/Profile.js]: not required!

File: src/components/Profile.js [hunk 2]
```diff
@@ -142,10 +143,11 @@ Profile.propTypes = {
     patchInfo: PropTypes.func.isRequired
 };
 
-const mapStateToProps = state => ({
+const mapStateToProps = (state) => ({
     userinfo: state.info.userinfo,
     userinfoerror: state.info.userinfoerror,
     userinfoid: state.info.userinfoid,
+
```
Comment (bismitaguha) [inline src/components/Profile.js]: not required!

File: src/components/Profile.js [hunk 3]
```diff
@@ -71,7 +72,7 @@ class Profile extends Component {
                         <Form>
                             <Form.Input
                                 name="name"
-                                value={this.state.name ? this.state.name : userinfo.name}
+                                value={this.state.name}
```
Comment (bismitaguha) [inline src/components/Profile.js]: Even when the name exists in the backend, this shows as empty when tried to change, whereas the already existent name should be displayed.

**Review by bismitaguha — CHANGES_REQUESTED** [review]

**Comment by bismitaguha** [comment]
@saideepesh000 I feel a probable fix would be using `this.setState()` to assign the incoming props object to the state variable.

File: src/components/Profile.js (hunk 3 above)
Comment (iamsriranga) [reply src/components/Profile.js]: > 
> 
> Even when the name exists in the backend, this shows as empty when tried to change, whereas the already existent name should be displayed.

@bismitaguha, actually for the first time when we load the site the name disappears, but from next time it's appearing.

**Comment by saideepesh000** [comment]
> 
> 
> @saideepesh000 I feel a probable fix would be using `this.setState()` to assign the incoming props object to the state variable.



> 
> 
> @saideepesh000 I feel a probable fix would be using `this.setState()` to assign the incoming props object to the state variable.

Ok, I'll try this once.

**Comment by saideepesh000** [comment]
@bismitaguha, I tried many ways they didn't work, this one worked, even when we refresh, the value is not disappearing.

**Comment by bismitaguha** [comment]
On the first load the already existent value is not present, but that is a major flaw.
try
```
this.setState({
    name: this.props.<variable_the_name_is_in>,
})
```
So in layman's term, it kind of gets copied in the state variable, Do this in `componentDidMount()` or before rendering the component.

**Comment by saideepesh000** [comment]
@bismitaguha, this time it's appearing, wait i'll send the gif.

**Comment by saideepesh000** [comment]
edit
@bismitaguha

**Comment by bismitaguha** [comment]
This looks fine as per gif. Push the code I will review it!!! @saideepesh000

**Comment by saideepesh000** [comment]
> 
> 
> This looks fine as per gif. Push the code I will review it!!! @saideepesh000

I already pushed it.

**Comment by saideepesh000** [comment]
@bismitaguha, I forgot to add this, "this.state.name?this.state.name:"

File: src/components/Profile.js [hunk 4]
```diff
@@ -107,7 +107,7 @@ class Profile extends Component {
                     <Form>
                         <Form.Input
                             name="name"
-                            value={this.state.name}
+                            value={this.state.name }
```
Comment (bismitaguha) [inline src/components/Profile.js]: space change not needed

File: src/components/Profile.js [hunk 5]
```diff
@@ -142,7 +142,7 @@ Profile.propTypes = {
     patchInfo: PropTypes.func.isRequired
 };
 
-const mapStateToProps = state => ({
+const mapStateToProps = (state) => ({
```
Comment (bismitaguha) [inline src/components/Profile.js]: why?

**Comment by bismitaguha** [comment]
It's working fine but the other changes are not necessary, just remove those!

File: src/components/Profile.js (hunk 5 above)
Comment (iamsriranga) [reply src/components/Profile.js]: xD, I added "ownProps", and forgot to remove those.

**Comment by bismitaguha** [comment]
> @bismitaguha, I forgot to add this, "this.state.name?this.state.name:"

Please do this also

**Comment by saideepesh000** [comment]
> 
> 
> It's working fine but the other changes are not necessary, just remove those!

I have a doubt, are username(Login.js) and name(Profile.js) same

**Comment by bismitaguha** [comment]
> > It's working fine but the other changes are not necessary, just remove those!
> 
> I have a doubt, are username(Login.js) and name(Profile.js) same

No, refer to the database design in the Wiki.

**Review by sidvenu — APPROVED** [review]
LGTM. @bismitaguha could you test this to make sure it's working properly?

**Review by bismitaguha — APPROVED** [review]

**Comment by bismitaguha** [comment]
@sidvenu @abha224 I guess this PR is ready to be merged.
